imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/processingFlow/FlowStep.java

```diff
@@ -1,42 +1,24 @@
 package pl.edu.uj.kimage.processingFlow;
 
-import pl.edu.uj.kimage.Plugin;
 import pl.edu.uj.kimage.eventbus.Event;
+import pl.edu.uj.kimage.eventbus.EventBus;
 
-public class FlowStep {
+public abstract class FlowStep {
     private final int stepId;
-    private final Plugin stepRelatedPlugin;
+    private final EventBus eventBus;
 
-    public FlowStep(int stepId, Plugin stepRelatedPlugin) {
+    public FlowStep(int stepId, EventBus eventBus) {
         this.stepId = stepId;
-        this.stepRelatedPlugin = stepRelatedPlugin;
+        this.eventBus = eventBus;
     }
 
     public int getStepId() {
         return stepId;
     }
 
-    public void processEvent(Event event)
-    {
-        stepRelatedPlugin.processEvent(event);
+    protected EventBus getEventBus() {
+        return eventBus;
```

Comment:
Is this really necessary? Why do you want to keep `EventBus` in `FlowStep`, that's weird.